src/RpcServer/RpcServer.SmartContract.cs

```diff
@@ -141,79 +150,108 @@ private static JObject ToJson(System.Collections.Generic.IEnumerable<DataCache.T
             return array;
         }
 
-        private static JObject ToJson(StackItem item, int max)
+        private static JObject ToJson(StackItem item, Session session)
         {
             JObject json = item.ToJson();
             if (item is InteropInterface interopInterface && interopInterface.GetInterface<object>() is IIterator iterator)
             {
-                JArray array = new();
-                while (max > 0 && iterator.Next())
-                {
-                    array.Add(iterator.Value(null).ToJson());
-                    max--;
-                }
-                json["iterator"] = array;
-                json["truncated"] = iterator.Next();
+                Guid id = Guid.NewGuid();
+                session.Iterators.Add(id, iterator);
+                json["id"] = id.ToString();
```

Comment:
The resulting JSON contains `id` for each iterator even if sessions are disabled. Is it the desired behaviour?